mypy/subtypes.py

```diff
@@ -209,6 +217,15 @@ def visit_partial_type(self, left: PartialType) -> bool:
         # This is indeterminate as we don't really know the complete type yet.
         raise RuntimeError
 
+    def visit_type_type(self, left: TypeType) -> bool:
+        right = self.right
+        if isinstance(right, TypeType):
+            return is_subtype(left.item, right.item)
+        if isinstance(right, CallableType):
+            return right.is_type_obj() and is_subtype(left.item, right.ret_type)
+        # XXX Others? Union, Any, TypeVar
```

Comment:
TypeType should also be a subtype of `builtins.type` so that you can use `isinstance` with an argument of type `Type[C]`.